modin/test/storage_formats/pandas/test_internals.py

```diff
@@ -2507,3 +2518,36 @@ def f2(i):
 
     hook = MetaList(f1.remote())[2]
     assert ray.get(f2.remote(hook)) == 3
+
+
+def test_remote_function():
+    def get_func():
+        @remote_function
+        def remote_func(arg):
+            return arg
+
+        return remote_func
+
+    def get_capturing_func(arg):
+        @remote_function
+        def remote_func():
+            return arg
+
+        return remote_func
+
+    if Engine.get() in ("Ray", "Unidist"):
+        from modin.core.execution.utils import _remote_function_cache
+
+        cache_len = len(_remote_function_cache)
+        assert get_func() is get_func()
```

Comment:
Why is this an assert?

Reply:
Sorry, what do you mean?

Reply:
I guess @arunjose696 asked what this assert is for, i.e., what does it suppose to check.

Reply:
That it returns the same object, from the cache.